### PR TITLE
Import all certificates from propagated bundles into Keycloak's truststore

### PR DESCRIPTION
**Pull request by mmorhun**
Signed-off-by: Mykola Morhun <mmorhun@redhat.com>

### What does this PR do?
Makes it possible to import several CA certificates from a single file into Keycloak's java trust store.

### What issues does this PR fix or reference?
https://github.com/eclipse/che/issues/18339

### How to test this PR?
1. Create a *.pem file that contains several root CA certificates
2. Create a configmap with single data field which contains the *.pem file
3. Add the configmap as trusted for Che
4. Check that all CA certs are added into Keycloak's trust store

### CODE REVIEW

File: pkg/deploy/identity-provider/deployment_keycloak.go [hunk 1]
```diff
@@ -714,3 +718,51 @@ func ProvisionKeycloakResources(deployContext *deploy.DeployContext) error {
 	_, err = util.K8sclient.ExecIntoPod(podToExec, keycloakProvisionCommand, "create realm, client and user", deployContext.CheCluster.Namespace)
 	return err
 }
+
+// getImportCABundleScript returns string which cntains bash function that imports ca-bundle into jks
+// The function has three arguments:
+// - absolute path to ca-bundle file
+// - absolute path to java keystore
+// - java keystore password
+func getImportCABundleScript() string {
```
Comment (davidfestal) [inline pkg/deploy/identity-provider/deployment_keycloak.go:727]: How do you manage the case where there are duplicate certificates (I assume we should be robust and ignore this error) ?
What about other errors when importing a certificate, Do we expect them to fail the start of the server ?

**Review by davidfestal — COMMENTED** [review]
What about the Che server ? Don't we add certificates in the java truststore in the Che Java deployment as well ? Does it need the same fix ?

File: pkg/deploy/identity-provider/deployment_keycloak.go (hunk 1 above)
Comment (mmorhun) [reply pkg/deploy/identity-provider/deployment_keycloak.go:727]: > How do you manage the case where there are duplicate certificates (I assume we should be robust and ignore this error) ?

When adding a certificate into trust store we pass unique alias for it. So if user provides identical certificates - both will be added to trust store without any error.

> What about other errors when importing a certificate, Do we expect them to fail the start of the server ?

Invalid certificates will not be imported, Keycloak and Che server will be started without invalid certificates

**Comment by mmorhun** [comment]
> What about the Che server ? Don't we add certificates in the java truststore in the Che Java deployment as well ? Does it need the same fix ?

@davidfestal already there: https://github.com/eclipse/che/pull/18504

File: pkg/deploy/identity-provider/deployment_keycloak.go (hunk 1 above)
Comment (davidfestal) [reply pkg/deploy/identity-provider/deployment_keycloak.go:727]: Thanks for the explanation.

**Comment by davidfestal** [comment]
> > What about the Che server ? Don't we add certificates in the java truststore in the Che Java deployment as well ? Does it need the same fix ?
> 
> @davidfestal already there: [eclipse/che#18504](https://github.com/eclipse/che/pull/18504)

Great ! Thanks !

**Comment by mmorhun** [comment]
@davidfestal do you have any other remarks?

File: pkg/deploy/identity-provider/deployment_keycloak.go [hunk 2]
```diff
@@ -714,3 +718,51 @@ func ProvisionKeycloakResources(deployContext *deploy.DeployContext) error {
 	_, err = util.K8sclient.ExecIntoPod(podToExec, keycloakProvisionCommand, "create realm, client and user", deployContext.CheCluster.Namespace)
 	return err
 }
+
+// getImportCABundleScript returns string which cntains bash function that imports ca-bundle into jks
```
Comment (AndrienkoAleksandr) [inline pkg/deploy/identity-provider/deployment_keycloak.go:722]: cntains > contains

**Review by AndrienkoAleksandr — APPROVED** [review]

**Pull request merged**